### PR TITLE
Support embedded code blocks in markdown; support astro

### PR DESCRIPTION
**Pull request by dsummersl**
## Summary

- Adds `.astro` file detection to `LANGUAGE_EXTENSIONS`
- Introduces `extract_astro_frontmatter` in `parse.py` to pull the TypeScript/JavaScript block from between the `---` delimiters, padding it with a leading blank line so all reported line numbers map exactly to the original file
- Adds `_parse_astro_file` which re-parses that extracted code as TypeScript, meaning **all existing TypeScript rules** (function/class/interface extraction, identifier anonymisation, loose-mode literal normalisation) apply for free with zero changes to the rest of the pipeline
- Falls back to HTML parsing for template-only Astro files that have no frontmatter
- Adds three fixture files (`one.astro`, `two.astro`, `template_only.astro`) and 13 tests covering frontmatter extraction edge cases, language detection, and region extraction with correct line numbers

## Design rationale

Astro files are *markdown-with-TypeScript-frontmatter* — the meaningful code lives in the `---` block; the HTML template is secondary for similarity detection. Rather than introducing a new `AstroConfig` and a bespoke Astro tree-sitter ruleset, we extract the frontmatter and hand it to the TypeScript parser. This keeps the change small and establishes a reusable pattern for other frontmatter-style template languages (MDX, Vue `<script setup>`, ERB, etc.): **extract the embedded code section → pad to preserve line numbers → re-parse under the host language**.

## Example

```astro
---                                   ← line 1  (opening delimiter, becomes blank padding)
import Layout from './Layout.astro';  ← line 2  ✓ reported correctly
const { title } = Astro.props;        ← line 3  ✓
                                      ← line 4
function buildTitle(t: string) {      ← line 5  ✓ extracted as a TypeScript function region
  return `Site - ${t}`;
}
---
<Layout title={buildTitle(title)} />
```

## Test plan

- [x] All 13 new Astro tests pass (`tests/pipeline/languages/test_astro.py`)
- [x] Full test suite green (109 tests)
- [ ] Manual smoke-test: `treepeat detect tests/fixtures/astro/ --ruleset loose` — should report similarity between `one.astro` and `two.astro` (structurally identical functions, different variable names)

https://claude.ai/code/session_01548vqF5cwxepgs2EpbZWVn

### CODE REVIEW

**Comment by dsummersl** [comment]
@copilot resolve the merge conflicts in this pull request

File: treepeat/pipeline/languages/__init__.py [hunk 1]
```diff
@@ -15,36 +16,36 @@
...
     "go": [".go"],
+    "html": [".html", ".htm"],
     "java": [".java"],
+    "javascript": [".js", ".jsx"],
     "kotlin": [".kt", ".kts"],
+    "markdown": [".md", ".markdown"],
+    "python": [".py"],
     "rust": [".rs"],
+    "sql": [".sql"],
+    "typescript": [".ts", ".tsx"],
```
[the remark this answers is not in the document]
Comment (dsummersl) [reply treepeat/pipeline/languages/__init__.py:48]: @copilot apply changes based on [this feedback](https://github.com/dsummersl/treepeat/pull/87#discussion_r3118380006)

File: treepeat/pipeline/languages/astro.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+class AstroConfig(LanguageConfig):
+    """Configuration for Astro (.astro) components.
+
+    Astro files consist of an optional TypeScript/JavaScript *frontmatter* block
+    (delimited by ``---`` on the first and last line) followed by an HTML-like
+    template.  The frontmatter is declared as a language-injection region so
+    that the shingler re-parses it as TypeScript and applies full TypeScript
+    normalization rules.  This ensures that identical TypeScript code in an
+    ``.astro`` file produces the same shingle hashes as the same code in a
+    ``.ts`` file, enabling cross-file similarity detection.
```
[the remark this answers is not in the document]
Comment (dsummersl) [reply treepeat/pipeline/languages/astro.py:15]: @copilot apply changes based on [this feedback](https://github.com/dsummersl/treepeat/pull/87#discussion_r3120257476)

File: treepeat/pipeline/languages/markdown.py [hunk 3]
```diff
@@ -1,14 +1,33 @@
...
+
+    Returns an empty string when no language tag is present; the injection
+    mechanism treats an empty string as "no injection".
+    """
+    for child in node.children:
+        if child.type == "info_string":
+            lang_text = source[child.start_byte : child.end_byte].decode("utf-8", errors="ignore").strip()
+            # Language identifiers may have extra text (e.g. "python title='foo'")
+            return lang_text.split()[0].lower() if lang_text else ""
+    return ""
```
[the remark this answers is not in the document]
Comment (dsummersl) [reply treepeat/pipeline/languages/markdown.py:40]: @copilot apply changes based on [this feedback](https://github.com/dsummersl/treepeat/pull/87#discussion_r3120257495)
